### DISCUSSION TITLE
Why next/image component has so much http requests

### DISCUSSION THREAD

**Discussion by hhhpw** [vercel/next.js#37786] [Help] [answered]
I have five images in my project and i want to show one of  these image 30ms.
my code it works.But I  found that repeated image requests appeared.
how to solve this problem? Is it bad code ?
```js


  let [imgSrc, setImgSrc] = useState("/static/a.png");
  useEffect(() => {
    const imgs = ["a", "b", "c", "d", "e", "f"];
    let i = 0;
    setInterval(() => {
      setImgSrc(`/static/${imgs[i]}.png`);
      i += 1;
      i = i > 5 ? (i = 0) : i;
    }, 30);
  }, []);



  <Image src={imgSrc} width={180} height={180} />

``` 
![image](https://user-images.githubusercontent.com/23467537/174256088-999c5bb3-2faa-47f1-9746-2b2d7c89d6f7.png)

**Comment by mhdalmajid** [comment, marked as answer]
this nothing to do with nextjs

its just the way you coded


### try this method

```jsx

import React from 'react';

const imgs = ['a', 'b', 'c', 'd', 'e', 'f'];

export default function App() {
  const [value, setValue] = React.useState(0);

  React.useEffect(() => {
    const interval = setInterval(() => {
      setValue(prev => {
        return prev === 5 ? 0 : prev + 1;
      });
    }, 1000);
    return () => clearInterval(interval);
  }, []);

  return <img src={`/static/${imgs[value]}.png`} width={180} height={180} />;
}

  ```

**Reply by hhhpw** [reply]
thanks.   I'm not good at react. I fixed it already.

**Reply by mhdalmajid** [reply]
happy to help , just ask when you need help

**Comment by icyJoseph** [comment]
Hi,

You could expect the browser caching to reduce the amount of requests, but every time a new image is mounted, the browser will start a request, even if it is cached (then it'll say something like Read from disk). This is not reliable at all. 

This could be fixed by tackling this from another perspective:

```tsx
import { useEffect, useState } from "react";

import type { NextPage } from "next";
import Image from "next/image";

const imgs = ["a", "b", "c", "d", "e", "f"];

const Home: NextPage = () => {
  const [pointer, setPointer] = useState(0);

  useEffect(() => {
    const interval = setInterval(() => {
      setPointer((prev) => (prev === imgs.length - 1 ? 0 : prev + 1));
    }, 30);

    return () => clearInterval(interval);
  }, []);

  const currentSrc = imgs[pointer];

  return (
    <div>
      {imgs.map((resource) => {
        const imgSrc = `/static/${resource}.png`;

        return (
          <div key={resource} hidden={currentSrc !== resource}>
            <Image src={imgSrc} width={180} height={180} alt={resource} />
          </div>
        );
      })}
    </div>
  );
};

export default Home;
```

This way, the images are loaded only once, and with CSS or, in this case, the [`hidden` attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/hidden), we can show one at the time to the user. 

30 ms is very fast though!

**Reply by hhhpw** [reply]
thank you very much

**Reply by icyJoseph** [reply]
How did you solve it?

**Reply by hhhpw** [reply]
like your methods with css attribute.  
```js
return () => clearInterval(interval);  // not works.the much requests exits yet.
```

**Reply by icyJoseph** [reply]
Nice :D should this be the accepted answer?

This part:

```ts
return () => clearInterval(interval)
```

Is not to limit the amount of requests, but to stop them if the page/component unmounts. 

React has something called StrictMode and in React 18 development mode that does (approx):

- mount
- unmount
- mount

You can read more about it here, https://reactjs.org/docs/strict-mode.html#ensuring-reusable-state, but this is pretty much necessary to help you see where concurrent rendering might trigger unwanted behaviour. If you are just learning React, it is best to keep this activated.

So if we don't clear the interval on unmount, we'll end up with a `setInterval` that's trying to set state on a component that no longer exists. Which makes the entire snippet I posted, necessary 😅
